src/components/YextEntityFieldSelector.tsx

```diff
@@ -42,41 +43,51 @@ export const YextEntityFieldSelector = <T extends Record<string, any>>(
     label: props.label,
     render: ({ field, value, onChange }: RenderProps) => {
       const filteredEntityFields = getFilteredEntityFields(props.filter);
+      const [entityFieldMode, setEntityFieldMode] = useState<boolean>(true);
```

Comment:
whether it's in entityFieldMode should be based on constantValueEnabled (line 14)

Reply:
hmmm constantValueEnabled isn't a prop received by the YextEntityFieldSelector?

Reply:
It should be on `value`... ideally `value` would be typed as `YextEntityField` -- not sure if we missed that or if it's not possible bc it's dynamically reading the Puck field

Reply:
oh value is set in the YextEntityFieldSelector though. So user can't initially pass in a value to it? But can change it in the UI after it loads?

Reply:
I'm not sure I fully understand the question, but `constantValueEnabled` is updated based on when the user clicks the button to enable constantValue mode or not. We should use that to determine what to render.

Reply:
Sounds good, that's what I did